src/utils/inverse_matrix.cpp

```diff
@@ -0,0 +1,45 @@
+//========================================================================================
```

Comment:
These matrix/vector related utils can be combined into one file.

Reply:
in one file now.